js/ui/viewSelector.js

```diff
@@ -146,6 +151,33 @@ const ViewsDisplayLayout = new Lang.Class({
             searchResultsBox.y2 = searchResultsBox.y1 + searchResultsHeight;
             this._searchResultsActor.allocate(searchResultsBox, flags);
         }
+    },
+
+    _updateCompositeMode: function() {
+        /* Given that the height of the widget varies according to the
+         * number of icons, we only check the width here.
+         */
+        this.compositeMode = Main.layoutManager.primaryMonitor &&
+                             Main.layoutManager.primaryMonitor.height < 600;
+    },
+
+    set compositeMode(mode) {
+        if (this._compositeMode != mode) {
```

Comment:
I would replace this with an early return, to reduce the indentation level below. Something like this:
```javascript
    if (this._compositeMode == mode)
        return;

    [ Rest of the code ]
```